docs/configuration.rst

```diff
@@ -449,17 +453,29 @@ remain available as described at the start of the :ref:`Configuration` section.
     other operation in your `Magpie` instance. The application will quit with an error if this value cannot be found.
     It is recommended that the developer configures every new instance with server-specific and strong credentials.
 
+    Prior versions also allowed modification of this value from the API and UI, which increased chances of having
+    out-of-sync definitions between the database and :term:`Configuration` files. This is not permitted anymore.
+    Changing this value should be accomplished by updating the :term:`Configuration` file and restarting the
+    :ref:`usage_webapp` or calling the :ref:`cli_helpers` to register changes.
+
+
 - ``MAGPIE_ADMIN_PASSWORD``
   .. no default since explicit value is now required
 
-  Password of the default 'administrator' :term:`User` generated by the application (see ``MAGPIE_ADMIN_USER`` details).
+  Password of the default 'administrator' :term:`User` generated by the application.
+
+  .. seealso::
+    ``MAGPIE_ADMIN_USER``
 
   .. versionchanged:: 2.0
-    Prior to this version, a default value was employed if this setting was not provided. Later `Magpie` version now
-    require an explicit definition of this parameter to avoid weak default configuration making the protected system
-    prone to easier breaches. This value **MUST** be defined before starting the application in order to resume to any
-    other operation in your `Magpie` instance. The application will quit with an error if this value cannot be found.
-    It is recommended that the developer configures every new instance with server-specific and strong credentials.
+    Default values definition and update during runtime for this parameter was modified to avoid problematic
+    configuration synchronization problems. See corresponding change details in above ``MAGPIE_ADMIN_USER``.
+
+  .. versionchanged:: 3.8
+    Prior to this version, changing only the ``MAGPIE_ADMIN_PASSWORD`` without modification of ``MAGPIE_ADMIN_USER``
+    was not handled. Following versions reapply the password systematically on restart to apply any changes of
+    credentials. Provided value must now also fulfill validation against standard password format requirements,
+    such as ``MAGPIE_PASSWORD_MIN_LENGTH``.
```

Comment:
Oh crap so now the default `qwerty` in `env.local` https://github.com/bird-house/birdhouse-deploy/blob/7ddfd5a587c2b4e6374e16283bca8924b70fb1e4/birdhouse/env.local.example#L14 will blow up?  

All existing instances that upgrade to this version of Magpie with the old `qwerty` or some other existing password less than 12 length will also blow up?

I hope you leave the existing password as-is and only enforce on password change?  Otherwise this will be a painful upgrade for all the existing servers out there.

Reply:
For all the users that where created via any configuration file, this was already the case. 
Only admin was not respecting it, as bypassed due to missing check during registration. 

I can update to check password and update only if different, but yes it will be effective as soon as it is changed.